pyGHDL/dom/_Translate.py

```diff
@@ -472,45 +467,75 @@ def GetGenericsFromChainedNodes(
-            yield GenericProcedureInterfaceItem.parse(generic)
-        elif kind == nodes.Iir_Kind.Interface_Function_Declaration:
-            yield GenericFunctionInterfaceItem.parse(generic)
+            genericConstant = GenericConstantInterfaceItem.parse(generic)
+
+            if nodes.Get_Has_Identifier_List(generic):
+                nextNode = nodes.Get_Chain(generic)
+                for nextGeneric in utils.chain_iter(nextNode):
+                    if nodes.Get_Subtype_Indication(nextGeneric) == nodes.Null_Iir:
+                        genericConstant.Identifiers.append(GetNameOfNode(nextGeneric))
+                    else:
+                        generic = nextGeneric
+                        break
+            else:
+                generic = nodes.Get_Chain(generic)
+
+            yield genericConstant
         else:
-            position = Position.parse(generic)
-            raise DOMException(
-                "Unknown generic kind '{kindName}'({kind}) in generic '{generic}' at {file}:{line}:{column}.".format(
-                    kind=kind,
-                    kindName=kind.name,
-                    generic=generic,
-                    file=position.Filename,
-                    line=position.Line,
-                    column=position.Column,
+            if kind == nodes.Iir_Kind.Interface_Type_Declaration:
+                yield GenericTypeInterfaceItem.parse(generic)
+            elif kind == nodes.Iir_Kind.Interface_Package_Declaration:
+                yield GenericPackageInterfaceItem.parse(generic)
+            elif kind == nodes.Iir_Kind.Interface_Procedure_Declaration:
+                yield GenericProcedureInterfaceItem.parse(generic)
+            elif kind == nodes.Iir_Kind.Interface_Function_Declaration:
+                yield GenericFunctionInterfaceItem.parse(generic)
+            else:
+                position = Position.parse(generic)
+                raise DOMException(
+                    "Unknown generic kind '{kindName}'({kind}) in generic '{generic}' at {file}:{line}:{column}.".format(
+                        kind=kind,
+                        kindName=kind.name,
+                        generic=generic,
+                        file=position.Filename,
+                        line=position.Line,
+                        column=position.Column,
+                    )
                 )
-            )
+
+            generic = nodes.Get_Chain(generic)
 
 
 @export
 def GetPortsFromChainedNodes(
     nodeChain: Iir,
 ) -> Generator[PortInterfaceItem, None, None]:
-    for port in utils.chain_iter(nodeChain):
+
+    port = nodeChain
+    while port != nodes.Null_Iir:
         kind = GetIirKindOfNode(port)
         if kind == nodes.Iir_Kind.Interface_Signal_Declaration:
             from pyGHDL.dom.InterfaceItem import PortSignalInterfaceItem
 
             portSignal = PortSignalInterfaceItem.parse(port)
 
+            if nodes.Get_Has_Identifier_List(port):
```

Comment:
@tgingold This solves how to handle multiple identifiers, right?

1. A manual iteration is needed
2. A lookahead is needed for certain node types (here a port signal)
3. If the subtype indication is Null, add this identifier to the existing currently created port.
4. If the subtype indication is present, a new port was found.

Am I right, that no other element is in between these nodes, so no need to check the kind of `nextPort`?

-----------------
In pyVHDLModel, it's not expanded to multiple ports. It's one port with a list of identifiers.